### ISSUE TITLE
Compiling one post

### ISSUE THREAD

**Issue by Jaymon** [Jaymon/bang#22]
This is an interesting one, and it has to do with how I architected the compiler.

if there are other directories in the post directory, those are considered "other directories" and so they will be copied over when the others list is ran through and copied (you can see Site.output() to see where this is done). 

This keeps duplicated work to a minimum (since posts won't copy everything over only to have others copy them over again or check if they exist) but also isn't incredibly elegant, because if we ever just want to compile one post, we would need to compile the post and any other directories would need to be ran through separately.

Basically, there is no good way to just compile one post completely right now. I think the solution would be to have posts take care of their trees, but that would require merging Directory and Post tighter together because right now Directory just goes through things and doesn't distinguish them, that would need to change because it would need to know it is in a Post or Aux or Other and act accordingly.

**Comment by Jaymon** [comment]
Directory does have `.is_post`, and `.is_aux` methods, that means Directory does have enough information to figure all this out, I would probably just need to have it figure everything out in `__iter__` and then change `Post.output` to handle other directories.

So the idea would be that if a directory gets classified as either post or aux, it will handle it tree from then on, but if a directory has neither classification it would be an other directory.

This could be as inefficient of just adding:

* Directory.posts()
* Directory.auxs()
* Directory.others()

And going through the directory tree 3 different times.
